### ISSUE TITLE
Sound does not work with ALSA

### ISSUE THREAD

**Issue by nixi-awoo** [ValveSoftware/Dota-2#653]
The sound stopped working after the last update. I am not sure whether it is a problem with my system or if there is an issue with the game. Other applications work quite well.

I have tried restarting my system, reinstalling ALSA utilities and recompiling the lastest kernel.

```
Game update: AppID 570 "Dota 2", ProcID 6623, IP 0.0.0.0:0
ERROR: ld.so: object '/home/nixi/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/nixi/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
pid 6625 != 6624, skipping destruction (fork without exec?)
ERROR: ld.so: object '/home/nixi/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/nixi/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Using breakpad crash handler
Setting breakpad minidump AppID = 570
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198014480840 [API loaded yes]
Steam_SetMinidumpSteamID:  Setting Steam ID:  76561198014480840
Setting breakpad minidump AppID = 373300
Fontconfig warning: "/home/nixi/Games/SteamLibrary/steamapps/common/dota 2 beta/game/core/panorama/fonts/conf.d/41-repl-os-win.conf", line 148: Having multiple values in <test> isn't supported and may not work as expected
Fontconfig warning: "/home/nixi/Games/SteamLibrary/steamapps/common/dota 2 beta/game/core/panorama/fonts/conf.d/41-repl-os-win.conf", line 160: Having multiple values in <test> isn't supported and may not work as expected
Installing breakpad exception handler for appid(gameoverlayui)/version(20151118165528)
Installing breakpad exception handler for appid(gameoverlayui)/version(1.0)
Installing breakpad exception handler for appid(gameoverlayui)/version(1.0)
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 70: non-double matrix element
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 70: non-double matrix element
Fontconfig warning: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 78: saw unknown, expected number
Installing breakpad exception handler for appid(gameoverlayui)/version(1.0)
ALSA lib confmisc.c:674:(snd_determine_driver) could not open control for card 0
ALSA lib conf.c:3328:(snd_config_hooks_call) function snd_config_hook_load_for_all_cards returned error: Invalid argument
ALSA lib confmisc.c:674:(snd_determine_driver) could not open control for card 0
ALSA lib conf.c:4241:(_snd_config_evaluate) function snd_func_card_driver returned error: Invalid argument
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4241:(_snd_config_evaluate) function snd_func_concat returned error: Invalid argument
ALSA lib confmisc.c:1251:(snd_func_refer) error evaluating name
ALSA lib conf.c:4241:(_snd_config_evaluate) function snd_func_refer returned error: Invalid argument
ALSA lib conf.c:4720:(snd_config_expand) Evaluate error: Invalid argument
ALSA lib pcm.c:2217:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.default
ALSA lib confmisc.c:674:(snd_determine_driver) could not open control for card 0
ALSA lib conf.c:4241:(_snd_config_evaluate) function snd_func_card_driver returned error: Invalid argument
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4241:(_snd_config_evaluate) function snd_func_concat returned error: Invalid argument
ALSA lib confmisc.c:1251:(snd_func_refer) error evaluating name
ALSA lib conf.c:4241:(_snd_config_evaluate) function snd_func_refer returned error: Invalid argument
ALSA lib conf.c:4720:(snd_config_expand) Evaluate error: Invalid argument
ALSA lib pcm.c:2217:(snd_pcm_open_noupdate) Unknown PCM default
AL lib: alsa.c:771: Could not open capture device 'default': Invalid argument
ALSA lib confmisc.c:674:(snd_determine_driver) could not open control for card 0
ALSA lib conf.c:4241:(_snd_config_evaluate) function snd_func_card_driver returned error: Invalid argument
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4241:(_snd_config_evaluate) function snd_func_concat returned error: Invalid argument
ALSA lib confmisc.c:1251:(snd_func_refer) error evaluating name
ALSA lib conf.c:4241:(_snd_config_evaluate) function snd_func_refer returned error: Invalid argument
ALSA lib conf.c:4720:(snd_config_expand) Evaluate error: Invalid argument
ALSA lib pcm.c:2217:(snd_pcm_open_noupdate) Unknown PCM default
AL lib: alsa.c:771: Could not open capture device 'default': Invalid argument
AL lib: oss.c:333: Could not open /dev/dsp: No such file or directory
ALSA lib confmisc.c:674:(snd_determine_driver) could not open control for card 0
ALSA lib conf.c:4241:(_snd_config_evaluate) function snd_func_card_driver returned error: Invalid argument
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4241:(_snd_config_evaluate) function snd_func_concat returned error: Invalid argument
ALSA lib confmisc.c:1251:(snd_func_refer) error evaluating name
ALSA lib conf.c:4241:(_snd_config_evaluate) function snd_func_refer returned error: Invalid argument
ALSA lib conf.c:4720:(snd_config_expand) Evaluate error: Invalid argument
ALSA lib pcm.c:2217:(snd_pcm_open_noupdate) Unknown PCM default
```

My system:

```
OS: Arch Linux 
Kernel: x86_64 Linux 4.2.5-1-ARCH
CPU: AMD Athlon II X4 630 @ 3.22GHz
GPU: AMD Radeon HD 6900 Series
RAM: 12020MiB

~ » lspci | grep -i audio
00:14.2 Audio device: Advanced Micro Devices, Inc. [AMD/ATI] SBx00 Azalia (Intel HDA)
01:00.1 Audio device: Advanced Micro Devices, Inc. [AMD/ATI] Cayman/Antilles HDMI Audio [Radeon HD 6900 Series]

~ » cat /proc/asound/cards
 0 [SB             ]: HDA-Intel - HDA ATI SB
                      HDA ATI SB at 0xfe024000 irq 16
 1 [HDMI           ]: HDA-Intel - HDA ATI HDMI
                      HDA ATI HDMI at 0xfdefc000 irq 28

~ » arecord -L
null
    Discard all samples (playback) or generate zero samples (capture)
default:CARD=SB
    HDA ATI SB, ALC889A Analog
    Default Audio Device
sysdefault:CARD=SB
    HDA ATI SB, ALC889A Analog
    Default Audio Device
front:CARD=SB,DEV=0
    HDA ATI SB, ALC889A Analog
    Front speakers
surround21:CARD=SB,DEV=0
    HDA ATI SB, ALC889A Analog
    2.1 Surround output to Front and Subwoofer speakers
surround40:CARD=SB,DEV=0
    HDA ATI SB, ALC889A Analog
    4.0 Surround output to Front and Rear speakers
surround41:CARD=SB,DEV=0
    HDA ATI SB, ALC889A Analog
    4.1 Surround output to Front, Rear and Subwoofer speakers
surround50:CARD=SB,DEV=0
    HDA ATI SB, ALC889A Analog
    5.0 Surround output to Front, Center and Rear speakers
surround51:CARD=SB,DEV=0
    HDA ATI SB, ALC889A Analog
    5.1 Surround output to Front, Center, Rear and Subwoofer speakers
surround71:CARD=SB,DEV=0
    HDA ATI SB, ALC889A Analog
    7.1 Surround output to Front, Center, Side, Rear and Woofer speakers
iec958:CARD=SB,DEV=0
    HDA ATI SB, ALC889A Digital
    IEC958 (S/PDIF) Digital Audio Output

```

**Comment by gdrewb-valve** [comment]
Reborn doesn't do any sound work directly, it uses SDL for sound on Linux, so Reborn has limited influence over what is happening with sound.  We didn't update SDL in the last Reborn update so it's unclear what would have changed. Do any other games or SDL applications work?

**Comment by gdrewb-valve** [comment]
Did anything else update on your system around the same time as the Reborn update?

**Comment by nixi-awoo** [comment]
I tested it with several games:

```
Guns of Icarus Online              - works
Counter-Strike: Global Offensive   - does not work
Half-Life 2                        - does not work
Arma 3                             - does not work
```

It is propably a problem with my system, even though I have no idea what problem it is and how to fix it.

**Comment by gdrewb-valve** [comment]
Did Steam itself update?  It's possible there's a Steam runtime issue.  You could try posting in steam-for-linux and maybe other people with similar behavior can comment.

**Comment by nixi-awoo** [comment]
Yes, Steam has updated recetly. I can not say for sure if that is the cause since I have not used my computer for almost 5 days. The Steam update could have been released at any time - before or after the Dota 2 update. I am going to check the steam-for-linux issue tracker.

**Comment by gdrewb-valve** [comment]
steam-for-linux#4139 seems close to what you're seeing.

**Comment by nixi-awoo** [comment]
It is indeed a Steam Runtime problem. An update for ALSA has been released (1.1.0), which is not compatible with Steam yet. There is an issue open on the steam-for-linux tracker:

https://github.com/ValveSoftware/steam-for-linux/issues/4139

I have fixed the problem by launching Steam with this parameter:

```
LD_PRELOAD=/usr/lib/libasound.so.2
```

Therefore, my entire launch command for Steam is:

```
sh -c 'LD_PRELOAD=/usr/lib/libasound.so.2 STEAM_FRAME_FORCE_CLOSE=1 steam' %U
```

I am sorry for not doing a proper research ^^

**Comment by gdrewb-valve** [comment]
OK, closing this out.
